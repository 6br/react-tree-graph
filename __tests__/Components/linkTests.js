import React from 'react';
import { shallow } from 'enzyme';

import Link from '../../Components/link';

const defaultProps = {
	source: {},
	target: {},
	keyProp: '',
	pathProps: {
		className: 'Link'
	},
	x1: 1,
	x2: 5,
	y1: 2,
	y2: 9
};

describe('<Link>', () => {
	test('renders correctly', () => {
		const tree = shallow(<Link {...defaultProps}/>);
		expect(tree).toMatchSnapshot();
	});

	test('click event has correct parameters', () => {
		const clickMock = jest.fn();
		const event = {};

		const props = {
			source: {
				data: {
					id: 'origin'
				}
			},
			target: {
				data: {
					id: 'target'
				}
			},
			keyProp: 'id',
			pathProps: {
				onClick: clickMock
			}
		};

		const tree = shallow(<Link {...defaultProps} {...props}/>);
		tree.find('path').simulate('click', event);

		expect(clickMock).toHaveBeenCalledTimes(1);
		expect(clickMock).toHaveBeenCalledWith('origin', 'target', event);
	});
	test('right click event has correct parameters', () => {
		const clickMock = jest.fn();
		const event = {};

		const props = {
			source: {
				data: {
					id: 'origin'
				}
			},
			target: {
				data: {
					id: 'target'
				}
			},
			keyProp: 'id',
			pathProps: {
				onContextMenu: clickMock
			}
		};

		const tree = shallow(<Link {...defaultProps} {...props}/>);
		tree.find('path').simulate('contextmenu', event);

		expect(clickMock).toHaveBeenCalledTimes(1);
		expect(clickMock).toHaveBeenCalledWith('origin', 'target', event);
	});
	test('clicking with no prop handler does nothing', () => {
		const props = {
			source: {
				data: {
					id: 'origin'
				}
			},
			target: {
				data: {
					id: 'target'
				}
			},
			keyProp: 'id'
		};

		const tree = shallow(<Link {...defaultProps} {...props}/>);
		tree.find('path').simulate('click');
	});
});