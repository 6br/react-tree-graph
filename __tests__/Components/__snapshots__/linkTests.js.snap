// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> renders correctly 1`] = `
<path
  className="Link"
  d="M2,1C5.5,1 5.5,5 9,5"
/>
`;

exports[`<Link> renders correctly with custom path 1`] = `
<path
  className="Link"
  d="M2,1 9,5"
/>
`;
